Build failure of msgpack_cpp11.cpp on g++6

```
g++ -DHAVE_CONFIG_H -I. -I..  -I../include -pthread -DMSGPACK_DISABLE_LEGACY_NIL -DMSGPACK_DISABLE_LEGACY_CONVERT -Wdate-time -D_FORTIFY_SOURCE=2  -O3 -Wall -Wextra -Wno-mismatched-tags -Werror -g -O2 -Wformat -Werror=format-security -std=c++11 -g -c -o msgpack_cpp11.o msgpack_cpp11.cpp
In file included from ../include/msgpack/adaptor/msgpack_tuple.hpp:18:0,
                 from ../include/msgpack/type.hpp:22,
                 from ../include/msgpack/object.hpp:839,
                 from ../include/msgpack.hpp:10,
                 from msgpack_cpp11.cpp:1:
../include/msgpack/adaptor/detail/cpp11_msgpack_tuple.hpp: In instantiation of 'static msgpack::v1::type::tuple<T, Args ...> msgpack::v1::MsgpackTupleAsImpl<T, Args>::as(const msgpack::v1::object&) [with T = no_def_con_composite; Args = {}]':
../include/msgpack/adaptor/detail/cpp11_msgpack_tuple.hpp:161:47:   required from 'static msgpack::v1::type::tuple<Args ...> msgpack::v1::MsgpackTupleAs<Args>::as(const msgpack::v1::object&) [with Args = {no_def_con_composite}]'
../include/msgpack/adaptor/detail/cpp11_msgpack_tuple.hpp:154:40:   required from 'static msgpack::v1::type::tuple<T, Args ...> msgpack::v1::MsgpackTupleAsImpl<T, Args>::as(const msgpack::v1::object&) [with T = no_def_con; Args = {no_def_con_composite}]'
../include/msgpack/adaptor/detail/cpp11_msgpack_tuple.hpp:161:47:   required from 'static msgpack::v1::type::tuple<Args ...> msgpack::v1::MsgpackTupleAs<Args>::as(const msgpack::v1::object&) [with Args = {no_def_con, no_def_con_composite}]'
../include/msgpack/adaptor/detail/cpp11_msgpack_tuple.hpp:154:40:   required from 'static msgpack::v1::type::tuple<T, Args ...> msgpack::v1::MsgpackTupleAsImpl<T, Args>::as(const msgpack::v1::object&) [with T = no_def_con; Args = {no_def_con, no_def_con_composite}]'
../include/msgpack/adaptor/detail/cpp11_msgpack_tuple.hpp:161:47:   required from 'static msgpack::v1::type::tuple<Args ...> msgpack::v1::MsgpackTupleAs<Args>::as(const msgpack::v1::object&) [with Args = {no_def_con, no_def_con, no_def_con_composite}]'
../include/msgpack/adaptor/detail/cpp11_msgpack_tuple.hpp:207:43:   required from 'msgpack::v1::type::tuple<Args ...> msgpack::v1::adaptor::as<msgpack::v1::type::tuple<Args ...>, typename std::enable_if<msgpack::v1::detail::all_of_imp<msgpack::v1::has_as<Args>::value ...>::value>::type>::operator()(const msgpack::v1::object&) const [with Args = {no_def_con, no_def_con, no_def_con_composite}; typename std::enable_if<msgpack::v1::detail::all_of_imp<msgpack::v1::has_as<Args>::value ...>::value>::type = void]'
../include/msgpack/object.hpp:559:37:   required from 'typename std::enable_if<msgpack::v1::has_as<T>::value, T>::type msgpack::v1::object::as() const [with T = msgpack::v1::type::tuple<no_def_con, no_def_con, no_def_con_composite>; typename std::enable_if<msgpack::v1::has_as<T>::value, T>::type = msgpack::v1::type::tuple<no_def_con, no_def_con, no_def_con_composite>]'
msgpack_cpp11.cpp:504:92:   required from here
../include/msgpack/adaptor/detail/cpp11_msgpack_tuple.hpp:154:43: error: could not convert 'msgpack::v1::type::tuple_cat(Tuples&& ...) [with Tuples = {msgpack::v1::type::tuple<no_def_con_composite>, msgpack::v1::type::tuple<>}; decltype (std::tuple_cat((forward<typename std::remove_reference<Tuples>::type::base>)(msgpack::v1::type::tuple_cat::args)...)) = std::tuple<no_def_con_composite>; typename std::remove_reference<Tuples>::type::base = <type error>](msgpack::v1::MsgpackTupleAs<>::as((* & o)))' from 'std::__tuple_cat_result<std::tuple<no_def_con_composite>, std::tuple<> >::__type {aka std::tuple<no_def_con_composite>}' to 'msgpack::v1::type::tuple<no_def_con_composite>'
             MsgpackTupleAs<Args...>::as(o));
                                           ^
```

--- a/include/msgpack/v1/adaptor/detail/cpp11_msgpack_tuple.hpp
+++ b/include/msgpack/v1/adaptor/detail/cpp11_msgpack_tuple.hpp
@@ -21,78 +21,29 @@ MSGPACK_API_VERSION_NAMESPACE(v1) {
 /// @endcond
 
 namespace type {
-    // tuple
-    using std::get;
-    using std::tuple_size;
-    using std::tuple_element;
-    using std::uses_allocator;
-    using std::ignore;
-    using std::swap;
 
-    template< class... Types >
-    class tuple : public std::tuple<Types...> {
-    public:
-        using base = std::tuple<Types...>;
+template <class... Args>
+inline tuple<Args...> make_tuple(Args&&... args) {
+    return tuple<Args...>(std::forward<Args>(args)...);
+}
 
-        using base::base;
+template<class... Args>
+inline tuple<Args&&...> forward_as_tuple (Args&&... args) noexcept {
+    return tuple<Args&&...>(std::forward<Args>(args)...);
+}
 
-        tuple() = default;
-        tuple(tuple const&) = default;
-        tuple(tuple&&) = default;
+template <class... Tuples>
+inline auto tuple_cat(Tuples&&... args) ->
+    decltype(
+        std::tuple_cat(std::forward<typename std::remove_reference<Tuples>::type::base>(args)...)
+    ) {
+    return std::tuple_cat(std::forward<typename std::remove_reference<Tuples>::type::base>(args)...);
+}
 
-        template<typename... OtherTypes>
-        tuple(tuple<OtherTypes...> const& other):base(static_cast<std::tuple<OtherTypes...> const&>(other)) {}
-        template<typename... OtherTypes>
-        tuple(tuple<OtherTypes...> && other):base(static_cast<std::tuple<OtherTypes...> &&>(other)) {}
-
-        tuple& operator=(tuple const&) = default;
-        tuple& operator=(tuple&&) = default;
-
-        template<typename... OtherTypes>
-        tuple& operator=(tuple<OtherTypes...> const& other) {
-            *static_cast<base*>(this) = static_cast<std::tuple<OtherTypes...> const&>(other);
-            return *this;
-        }
-        template<typename... OtherTypes>
-        tuple& operator=(tuple<OtherTypes...> && other) {
-            *static_cast<base*>(this) = static_cast<std::tuple<OtherTypes...> &&>(other);
-            return *this;
-        }
-
-        template< std::size_t I>
-        typename tuple_element<I, base >::type&
-        get() & { return std::get<I>(*this); }
-
-        template< std::size_t I>
-        typename tuple_element<I, base >::type const&
-        get() const& { return std::get<I>(*this); }
-
-        template< std::size_t I>
-        typename tuple_element<I, base >::type&&
-        get() && { return std::get<I>(*this); }
-    };
-
-    template <class... Args>
-    inline tuple<Args...> make_tuple(Args&&... args) {
-        return tuple<Args...>(args...);
-    }
-
-    template<class... Args>
-    inline tuple<Args&&...> forward_as_tuple (Args&&... args) noexcept {
-        return tuple<Args&&...>(std::forward<Args>(args)...);
-    }
-
-    template <class... Tuples>
-    inline auto tuple_cat(Tuples&&... args) ->
-        decltype(
-            std::tuple_cat(std::forward<typename std::remove_reference<Tuples>::type::base>(args)...)
-        ) {
-        return std::tuple_cat(std::forward<typename std::remove_reference<Tuples>::type::base>(args)...);
-    }
-    template <class... Args>
-    inline tuple<Args&...> tie(Args&... args) {
-        return tuple<Args&...>(args...);
-    }
+template <class... Args>
+inline tuple<Args&...> tie(Args&... args) {
+    return tuple<Args&...>(args...);
+}
 } // namespace type
 
 // --- Pack from tuple to packer stream ---

--- a/include/msgpack/v1/adaptor/detail/cpp11_msgpack_tuple_decl.hpp
+++ b/include/msgpack/v1/adaptor/detail/cpp11_msgpack_tuple_decl.hpp
@@ -32,7 +32,45 @@ namespace type {
     using std::swap;
 
     template< class... Types >
-    class tuple;
+    class tuple : public std::tuple<Types...> {
+    public:
+        using base = std::tuple<Types...>;
+
+        tuple(tuple const&) = default;
+        tuple(tuple&&) = default;
+
+        template<typename... OtherTypes>
+        tuple(OtherTypes&&... other):base(std::forward<OtherTypes>(other)...) {}
+
+        template<typename... OtherTypes>
+        tuple(tuple<OtherTypes...> && other):base(static_cast<std::tuple<OtherTypes...>>(other)) {}
+
+        tuple& operator=(tuple const&) = default;
+        tuple& operator=(tuple&&) = default;
+
+        template<typename... OtherTypes>
+        tuple& operator=(tuple<OtherTypes...> const& other) {
+            *static_cast<base*>(this) = static_cast<std::tuple<OtherTypes...> const&>(other);
+            return *this;
+        }
+        template<typename... OtherTypes>
+        tuple& operator=(tuple<OtherTypes...> && other) {
+            *static_cast<base*>(this) = static_cast<std::tuple<OtherTypes...> &&>(other);
+            return *this;
+        }
+
+        template< std::size_t I>
+        typename tuple_element<I, base >::type&
+        get() & { return std::get<I>(*this); }
+
+        template< std::size_t I>
+        typename tuple_element<I, base >::type const&
+        get() const& { return std::get<I>(*this); }
+
+        template< std::size_t I>
+        typename tuple_element<I, base >::type&&
+        get() && { return std::get<I>(*this); }
+    };
 
     template <class... Args>
     tuple<Args...> make_tuple(Args&&... args);

